client/trino-cli/src/main/java/io/trino/cli/QueryRunner.java

```diff
@@ -72,7 +81,9 @@ public QueryRunner(
             Optional<String> kerberosConfigPath,
             Optional<String> kerberosKeytabPath,
             Optional<String> kerberosCredentialCachePath,
-            boolean kerberosUseCanonicalHostname)
+            boolean kerberosUseCanonicalHostname,
+            Optional<String> externalAuthenticationTimeout,
+            String redirectHandlerMethod)
```

Comment:
`externalAuthenticationTimeout` has more responsibilities than `redirectHandlerMethod` and it only because private method has such implementation.
https://github.com/trinodb/trino/pull/7054/commits/b42465a057d96019a15731248addf47a2c3843f0#diff-94fce3babe25c57f3d3655748005429ce683acb7602a0226d71cb3a297a8e17fR181

May be it worth to propagate `externalAuthentication` to this method too?

Consider wrapping `externalAuthenticationTimeout` and `redirectHandlerMethod` to `externalAuthentication` object.